src/ngraph/op/fused/space_to_batch.cpp

```diff
@@ -0,0 +1,201 @@
+    {
+        axes_order.push_back(j);
+    }
+
+    flat_node = builder::opset1::reorder_axes(flat_node, axes_order);
+    Shape squeezed_shape;
+    int64_t prod = 1;
+    for (auto& el : block_values)
+    {
+        prod *= el;
+    }
+
+    //    y = reshape(x'', [batch * B_1 * ... * B_{N - 1}, (D_1 + P_1) / B_1, (D_2 + P_2) / B_2, ...
+    //    ,
+    //      (D_{N - 1} + P_{N - 1}) / B_{N - 1}])
+    squeezed_shape.push_back(out_shape.at(0) * prod);
+    for (size_t i = 1; i < block_values.size(); ++i)
+    {
+        squeezed_shape.push_back(out_shape.at(i) / block_values.at(i));
+    }
+    flat_node = builder::opset1::reshape(flat_node, squeezed_shape);
+
+    return NodeVector{flat_node};
+}
+
+void ngraph::op::v1::SpaceToBatch::pre_validate_and_infer_types()
+{
+    PartialShape data_pshape = get_input_partial_shape(0);
+
+    auto data = input_value(0);
+    auto block = input_value(1);
+    auto crops_begin = input_value(2);
+    auto crops_end = input_value(3);
+    NGRAPH_CHECK(block.get_node_shared_ptr()->is_constant(),
+                 "block_shape input node is expected to be a static constant");
+
+    NGRAPH_CHECK(crops_begin.get_node_shared_ptr()->is_constant(),
+                 "crops_begin input node is expected to be a static constant");
+
+    NGRAPH_CHECK(crops_end.get_node_shared_ptr()->is_constant(),
+                 "crops_end input node is expected to be a static constant");
+
+    const auto& data_type = get_input_element_type(0);
+    const auto& block_shape_type = get_input_element_type(1);
+    const auto& crops_begin_type = get_input_element_type(2);
+    const auto& crops_end_type = get_input_element_type(3);
+    NODE_VALIDATION_CHECK(this,
+                          block_shape_type == element::i32 || block_shape_type == element::i64,
+                          "block_shape element type must be either int64_t or int32_t but got (",
+                          block_shape_type,
+                          ").");
+
+    NODE_VALIDATION_CHECK(this,
+                          crops_begin_type == element::i32 || crops_begin_type == element::i64,
+                          "crops_begin element type must be either int64_t or int32_t but got (",
+                          crops_begin_type,
+                          ").");
+
+    NODE_VALIDATION_CHECK(this,
+                          crops_end_type == element::i32 || crops_end_type == element::i64,
+                          "crops_end element type must be either int64_t or int32_t but got (",
+                          crops_end_type,
+                          ").");
+
+    if (data_pshape.is_dynamic())
+    {
+        set_output_type(0, data_type, PartialShape::dynamic());
+    }
+}
+
+std::shared_ptr<Node>
+    ngraph::op::v1::SpaceToBatch::copy_with_new_args(const ngraph::NodeVector& new_args) const
+{
+    if (new_args.size() != 4)
+    {
+        throw ngraph_error("Incorrect number of new arguments");
+    }
```

Comment:
Please use this helper here as well.
`check_new_args_count(this, new_args);`

Reply:
fixed